package.json

```diff
@@ -12,8 +12,7 @@
     "test": "npm run build && nyc mocha --exit --timeout 20000 tests/**/*.test.js",
     "test:debug": "npm run build:dev && mocha -r source-map-support/register --inspect-brk --exit tests/**/*.test.js",
     "test:debug:vscode": "mocha -r source-map-support/register --exit tests/**/*.test.js",
-    "coverage": "nyc report --reporter=text-lcov | coveralls",
-    "coverage:html": "nyc --reporter=html mocha --exit --timeout 20000 tests/**/*.test.js",
+    "coverage": "nyc report --reporter=html --check-coverage --lines 50",
```

Comment:
IMO 85% would be a well suited threshold

Reply:
For now I am troubleshooting, 85 is low I would prefer 95

Reply:
Then you need to write tests for the worker classes 😛

Reply:
I will into this PR yes :D 
I need to figure out more on sonar with coverage( I am close to find a solution not sure if I will fix it today or tomorrow) and then I will write tests